### ISSUE TITLE
ParamSpec doesn't work as string

### ISSUE THREAD

**Issue by Agalin** [microsoft/pyright#1909]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Pyright doesn't consider lazy (string) annotations of  `.args` and `.kwargs` to be valid. Makes it hard to use in Python 3.6 with generics as you need some runtime workarounds for ParamSpec to be used with Generic and those workarounds break usage of `.args` and `.kwargs`.

**To Reproduce**
Wrap `P.args` or `P.kwargs` in string.

**Expected behavior**
No errors.

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")


def d1(f: Callable[P, None]):
    def wrapper(*args: P.args, **kwargs: P.kwargs):
        ...

def d2(f: Callable[P, None]):
    def wrapper(*args: "P.args", **kwargs: "P.kwargs"):
        ...

def d3(f: Callable[P, None]):
    def wrapper(*args: P.args, **kwargs: "P.kwargs"):
        ...

```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

Pylance 2021.5.3 from VSCode, pyright 1.1.143 from shell.

**Additional context**
Pyright output:
```
C:\Users\soloch\Desktop\pyright_demo\demo2.py
  demo2.py:12:25 - error: "args" member of ParamSpec is valid only when used with *args parameter
  demo2.py:12:45 - error: "kwargs" member of ParamSpec is valid only when used with **kwargs parameter
  demo2.py:16:43 - error: "kwargs" member of ParamSpec is valid only when used with **kwargs parameter
  demo2.py:16:24 - error: "args" and "kwargs" members of ParamSpec must both appear within a function signature
4 errors, 0 warnings, 0 infos
```

**Comment by Agalin** [comment]
It's PROBABLY unrelated but note that `*args` is incorrectly (even for `d1` case) resolved as `args: tuple[P@d1, ...]` while it should be `args: P@d1` as `P.args` describes the whole tuple, not a single value in it.

**Comment by erictraut** [comment]
Thanks for the bug report.

The `P.args` and `P.kwargs` annotation expressions are unlike any others in the type system, so they are handled with inelegant special-case logic. That explains why you're seeing some oddities here.

I've fixed the problem that prevents you from using `P.args` and `P.kwargs` in quoted annotations. The other issue you reported is largely cosmetic and is more difficult to fix, so I'm going to leave it as is for now. If you see a place where it's causing functional problems, let me know, and I'll prioritize it accordingly.

**Comment by Agalin** [comment]
I don't see any functional issue with that one as of now.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.144, which I just published. It will also be in the next release of pylance.
